std/path.d

```diff
@@ -4032,20 +4035,15 @@ version(unittest) import std.process : environment;
         if (oldHome !is null) environment["HOME"] = oldHome;
         else environment.remove("HOME");
 
-        // Test user expansion for root, no /root on Android
-        version (OSX)
-        {
-            assert(expandTilde("~root") == "/var/root", expandTilde("~root"));
-            assert(expandTilde("~root/") == "/var/root/", expandTilde("~root/"));
-        }
-        else version (Android)
-        {
-        }
-        else
-        {
-            assert(expandTilde("~root") == "/root", expandTilde("~root"));
-            assert(expandTilde("~root/") == "/root/", expandTilde("~root/"));
-        }
+        immutable user = environment.get("USER");
+        immutable tildeUser = "~" ~ user;
```

Comment:
I just tested this on my Android phone and this doesn't seem to work there.
`echo ~$USER` evaluates to `~shell`

Maybe simply use `version(Android) {} else` in front of this block?

CC @joakim-noah

Reply:
Yes I thought about Android but didn't know how to test it locally in a fast way so I hoped the auto tester will do it.
Thanks for testing it manually.
Since it is not working I will add the Android block as it was before.

Reply:
> Yes I thought about Android but didn't know how to test it locally in a fast way so I hoped the auto tester will do it.

If you have an Android phone, you can reproduce this by enabling USB debugging, plugging in your phone, running `adb shell` on your PC, and typing `echo ~$USER`.

Reply:
As DMD can't compile for arm, we don't have a CI testing for it. Most of the Android work is actually based on @joakim-noah's hard work at LDC. With LDC being so up to date, we should be able to incorporate their diff and run the Phobos testsuite directly with LDC too. Well that's a goal for 2018...

Reply:
Ok little misunderstanding. I thought you ran the actual test.
Even if the user on Android is shell and there is no home directory configured the test should not fail.
I already tested that case by creating a user without home directory. (useradd -r testuser)